test/Microsoft.AspNetCore.Mvc.TestCommon/CommonResourceInvokerTest.cs

```diff
@@ -1713,6 +1733,318 @@ public abstract class CommonResourceInvokerTest
                 Times.Never());
         }
 
+        [Fact]
+        public async Task InvokeAction_AuthorizationFilterShortCircuit_InvokesAlwaysRunResultFilter()
+        {
+            // Arrange
+            var resultFilter = new Mock<IAlwaysRunResultFilter>(MockBehavior.Strict);
+            resultFilter.Setup(f => f.OnResultExecuting(It.IsAny<ResultExecutingContext>()))
+                .Callback<ResultExecutingContext>(c => Assert.Same(Result, c.Result))
+                .Verifiable();
+            resultFilter.Setup(f => f.OnResultExecuted(It.IsAny<ResultExecutedContext>()))
+                .Callback<ResultExecutedContext>(c => Assert.Same(Result, c.Result))
+                .Verifiable();
+
+            var authorizationFilter = new Mock<IAuthorizationFilter>(MockBehavior.Strict);
```

Comment:
Should you try out a similar test but with IAsyncAuthorizationFilter too? A theory maybe. Applies to other tests here too.

Reply:
Wouldn't hurt to change a test or two to use async filters, but I don't think we want to have these tests be combinatorial.